tests/test_oneclick.py

```diff
@@ -0,0 +1,295 @@
+# encoding: utf-8
+
+from nose.tools import (
+    assert_raises_regexp,
+    eq_, 
+    set_trace,
+)
+
+import datetime
+import json
+import os
+
+from classifier import Classifier
+from coverage import CoverageFailure
+
+from model import (
+    Contributor,
+    DataSource, 
+    DeliveryMechanism,
+    Edition,
+    Identifier,
+    Hyperlink,
+    #LicensePool,
+    Representation,
+    Subject,
+)
+
+from oneclick import (
+    OneClickAPI,
+    MockOneClickAPI,
+    OneClickBibliographicCoverageProvider,
+    OneClickRepresentationExtractor,
+)
+
+from util.http import (
+    RemoteIntegrationException,
+    HTTP,
+)
+
+from . import DatabaseTest
+from scripts import RunCoverageProviderScript
+from testing import MockRequestsResponse
+
+
+class OneClickTest(DatabaseTest):
+
+    def setup(self):
+        super(OneClickTest, self).setup()
+        self.api = MockOneClickAPI(self._db)
+        base_path = os.path.split(__file__)[0]
+        self.resource_path = os.path.join(base_path, "files", "oneclick")
+
+    def get_data(self, filename):
+        # returns contents of sample file as string and as dict
+        path = os.path.join(self.resource_path, filename)
+        data = open(path).read()
+        return data, json.loads(data)
+
+
+class TestOneClickAPI(OneClickTest):
+
+    def test_create_identifier_strings(self):
+        identifier = self._identifier()
+        values = OneClickAPI.create_identifier_strings(["foo", identifier])
+        eq_(["foo", identifier.identifier], values)
+
+
+    def test_availability_exception(self):
+        api = MockOneClickAPI(self._db)
+        api.queue_response(500)
+        assert_raises_regexp(
+            RemoteIntegrationException, "Bad response from www.oneclickapi.testv1/libraries/library_id_123/search: Got status code 500 from external server, cannot continue.", 
+            api.get_all_available_through_search
+        )
+
+    def test_search(self):
+        api = MockOneClickAPI(self._db)
+        datastr, datadict = self.get_data("response_search_one_item_1.json")
+        api.queue_response(status_code=200, content=datastr)
+
+        response = api.search(mediatype='ebook', author="Alexander Mccall Smith", title="Tea Time for the Traditionally Built")
+        response_dictionary = response.json()
+        eq_(1, response_dictionary['pageCount'])
+        eq_(u'Tea Time for the Traditionally Built', response_dictionary['items'][0]['item']['title'])
+
+    def test_get_all_available_through_search(self):
+        api = MockOneClickAPI(self._db)
+        datastr, datadict = self.get_data("response_search_five_items_1.json")
+        api.queue_response(status_code=200, content=datastr)
+
```

Comment:
We discussed in person the conundrum of why this test works given your get_all_available_through_search implementation.